app/assets/javascripts/discourse/app/mixins/card-contents-base.js

```diff
@@ -365,4 +359,14 @@ export default Mixin.create({
 
     return true;
   },
+
+  @bind
+  _escListener(event) {
+    if (this.visible && event.key === "Escape") {
+      this._close();
+      this.cardTarget?.focus();
+      return;
+    }
+    return true;
```

Comment:
I don't think `return true;` is useful here, it can cancel keyboard events while the user card is visible.